### PR TITLE
sunpy.timeseries python file review

### PR DESCRIPTION
**Pull request by nabobalis**
Time for TimeSeries. 

Why does the bot hate me?

### CODE REVIEW

File: sunpy/timeseries/metadata.py [hunk 1]
```diff
@@ -450,25 +463,29 @@ def values(self):
...
         for metatuple in self.metadata:
             if end < metatuple[0].end:
-               end = metatuple[0].end
+                end = metatuple[0].end
         return TimeRange(start, end)
 
     def _remove_columns(self, colnames):
-        """Removes the given column/s from the TimeSeriesMetaData object.
+        """
+        Removes the given column(s) from the
```
Comment (Cadair) [inline sunpy/timeseries/metadata.py]: no line break here?

File: sunpy/timeseries/sources/__init__.py [hunk 2]
```diff
@@ -1,9 +1,19 @@
...
+from sunpy.timeseries.sources.eve import EVESpWxTimeSeries
+from sunpy.timeseries.sources.fermi_gbm import GBMSummaryTimeSeries
+from sunpy.timeseries.sources.goes import XRSTimeSeries
+from sunpy.timeseries.sources.lyra import LYRATimeSeries
+from sunpy.timeseries.sources.noaa import NOAAIndicesTimeSeries, NOAAPredictIndicesTimeSeries
+from sunpy.timeseries.sources.norh import NoRHTimeSeries
+from sunpy.timeseries.sources.rhessi import RHESSISummaryTimeSeries
 
-from . import eve, rhessi, goes, lyra, noaa, norh, fermi_gbm
+__all__ = ['EVESpWxTimeSeries', 'XRSTimeSeries', 'NOAAIndicesTimeSeries',
```
Comment (Cadair) [inline sunpy/timeseries/sources/__init__.py]: Do we need this? We are not excluding anything from the namespace, so we probably don't really need it.

File: sunpy/timeseries/sources/goes.py [hunk 3]
```diff
@@ -1,36 +1,32 @@
-# -*- coding: utf-8 -*-
```
Comment (Cadair) [inline sunpy/timeseries/sources/goes.py]: Given this file has unicode characters in it, we should probably keep the coding? I can't remember what the best practice on this is.

File: sunpy/timeseries/sources/noaa.py [hunk 4]
```diff
@@ -1,35 +1,34 @@
...
+"""
 from collections import OrderedDict
+
+import numpy as np
 from matplotlib import pyplot as plt
 from pandas.io.parsers import read_csv
-import numpy as np
 
-from sunpy.timeseries.timeseriesbase import GenericTimeSeries
+from astropy import units as u
```
Comment (Cadair) [inline sunpy/timeseries/sources/noaa.py]:
```suggestion
import astropy.units as u
```

File: sunpy/timeseries/sources/rhessi.py [hunk 5]
```diff
@@ -1,16 +1,19 @@
...
+This module provies a RHESSI `~sunpy.timeseries.TimeSeries` source.
+"""
 import datetime
+from collections import OrderedDict
+
 import matplotlib.dates
 import matplotlib.pyplot as plt
 from pandas import DataFrame
 
+from astropy import units as u
```
Comment (Cadair) [inline sunpy/timeseries/sources/rhessi.py]:
```suggestion
import astropy.units as u
```

File: sunpy/timeseries/metadata.py (hunk 1 above)
Comment (nabobalis) [reply sunpy/timeseries/metadata.py]: Doc Formater got there. I can fix it. Noticed another mistake.

File: sunpy/timeseries/sources/goes.py (hunk 3 above)
Comment (nabobalis) [reply sunpy/timeseries/sources/goes.py]: I recall if its python3, its set to that encoding by default.

Comment (Cadair) [reply sunpy/timeseries/sources/goes.py]: That does seem to be the case.

File: sunpy/timeseries/metadata.py [hunk 6]
```diff
@@ -82,29 +95,27 @@ def __init__(self, meta=None, timerange=None, colnames=None, **kwargs):
                 self.metadata.append((timerange, colnames, MetaDict()))
             elif isinstance(timerange, TimeRange):
                 self.metadata.append((timerange, [], MetaDict()))
-                warnings.warn("No time range given for metadata therefore metadata can't be linked to columns.",
-                              SunpyUserWarning)
+                warnings.warn("No time range given for metadata. This will mean the metadata can't be linked to columns in data.", SunpyUserWarning)
```
Comment (Cadair) [inline sunpy/timeseries/metadata.py]: Line length?

File: sunpy/timeseries/metadata.py [hunk 7]
```diff
@@ -520,30 +534,29 @@ def _validate_meta(self, meta):
...
+            Metadata dictionaries and column lists will be truncated if this is small.
+            Defaults to 10.
+        width : `int`, optional
+            The number of characters wide to make the entire table. Defaults to 99.
         """
         # Parameters
         colspace = ' | '
         liswidths = (26, 15, width-2-2*len(colspace) - 26 - 15)
-        colheadings = '|' + 'TimeRange'.ljust(100)[:liswidths[0]] + colspace + 'Columns'.ljust(100)[:liswidths[1]] + colspace + 'Meta'.ljust(100)[:liswidths[2]]  + '|'
+        colheadings = '|' + 'TimeRange'.ljust(100)[:liswidths[0]] + colspace + 'Columns'.ljust(100)[:liswidths[1]] + colspace + 'Meta'.ljust(100)[:liswidths[2]] + '|'
```
Comment (Cadair) [inline sunpy/timeseries/metadata.py]: line length

**Comment by nabobalis** [comment]
tox-conda seems to be broken right now for the figure and conda builds.
Time out issues on the hmi.py example.

**Comment by nabobalis** [comment]
```
===================================================================================== FAILURES =====================================================================================
___________________________________________________________________ [doctest] docs/guide/acquiring_data/hek.rst ____________________________________________________________________
[gw1] linux -- Python 3.7.2 /home/vsts/work/1/s/.tox/py37-online/bin/python
196 
197     >>> result = client.search( hek.attrs.Time(tstart,tend), hek.attrs.EventType(event_type), hek.attrs.FRM.Name == 'SSW Latest Events')  # doctest: +REMOTE_DATA
198     >>> len(result)  # doctest: +REMOTE_DATA
199     2
200 
201 We can also retrieve all the entries in the time range which were not
202 made by SSW Latest Events with the following command:
203 
204     >>> result = client.search( hek.attrs.Time(tstart,tend), hek.attrs.EventType(event_type),hek.attrs.FRM.Name != 'SSW Latest Events')  # doctest: +REMOTE_DATA
205     >>> len(result)  # doctest: +REMOTE_DATA
Expected:
    17
Got:
    19

```

wut

**Comment by Cadair** [comment]
oh good the number of HEK results have changed :man_facepalming:

**Pull request merged**